homeassistant/components/baf/config_flow.py

```diff
@@ -0,0 +1,117 @@
+"""Config flow for baf."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from aiobafi6 import Device, Service
+from aiobafi6.discovery import PORT
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, RUN_TIMEOUT
+from .models import BAFDiscovery
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_try_connect(ip_address: str) -> Device:
+    """Validate we can connect to a device."""
+    device = Device(Service(ip_addresses=[ip_address], port=PORT))
+    run_future = device.async_run()
+    try:
+        await asyncio.wait_for(device.async_wait_available(), timeout=RUN_TIMEOUT)
+    except asyncio.TimeoutError as ex:
+        raise CannotConnect from ex
+    finally:
+        run_future.cancel()
+    return device
+
+
+class BAFFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle BAF discovery config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the BAF config flow."""
+        self.discovery: BAFDiscovery | None = None
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+        properties = discovery_info.properties
+        ip_address = discovery_info.host
+        uuid = properties["uuid"]
+        model = properties["model"]
+        name = properties["name"]
+        await self.async_set_unique_id(uuid, raise_on_progress=False)
+        self._abort_if_unique_id_configured(updates={CONF_IP_ADDRESS: ip_address})
```

Comment:
We've recently had a few cases where we had to guard this to be IPv4 only. Should we do that here too?

Reply:
Good call